lib/domain.js

```diff
@@ -27,8 +27,13 @@ Object.defineProperty(process, 'domain', {
   }
 });
 
+// it's possible to enter one domain while already inside
+// another one.  the stack is each entered domain.
+var stack = [];
+exports._stack = stack;
+
 // let the process know we're using domains
-const _domain_flag = process._setupDomainUse(_domain);
+const _domain_flag = process._setupDomainUse(_domain, stack);
```

Comment:
Generally, yes, although it seems it's not enforced by the linter. Are you saying that I should fix `_domain_flag`'s name to be camel cased? This is not an addition from this PR, so i'd rather focus on the changes that this PR makes and fix that later.